zenith_utils/src/pq_proto.rs

```diff
@@ -798,3 +813,156 @@ impl<'a> BeMessage<'a> {
         Ok(())
     }
 }
+
+// Zenith extension of postgres replication protocol
+// See ZENITH_STATUS_UPDATE_TAG_BYTE
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+pub struct ZenithFeedback {
+    // Last known size of the timeline. Used to enforce timeline size limit.
+    pub current_instance_size: u64,
+    // Parts of StandbyStatusUpdate we resend to compute via safekeeper
+    pub ps_writelsn: u64,
+    pub ps_applylsn: u64,
+    pub ps_flushlsn: u64,
+    pub ps_replytime: SystemTime,
+}
+pub const ZENITH_FEEDBACK_FIELDS_NUMBER: u8 = 5;
+
+impl ZenithFeedback {
+    pub fn empty() -> ZenithFeedback {
```

Comment:
This shouldn't be needed it enum is used to store it in the `ReplicaState`. In its current form It looks like a placeholder value which possible can be misinterpreted, so I think solution with enum is preferable

Reply:
We do need some default feedback values to fill AppendResponse before any real feedback is received from pageserver's walreceiver.
These 0 values are interpreted as invalid by Postgres walsender. See comment below